IdentityCore/src/requests/MSIDTokenResponseHandler.m

```diff
@@ -89,6 +92,16 @@ - (void)handleTokenResponse:(MSIDTokenResponse *)tokenResponse
         return;
     }
     
+#if !AD_BROKER
+    // Save client flights if available
+    if (tokenResponse.clientFlights)
+    {
+        MSIDBrokerFlightProvider *flightProvider = [[MSIDBrokerFlightProvider alloc] initWithBase64EncodedFlightsPayload:tokenResponse.clientFlights];
+        
+        [MSIDFlightManager sharedInstance].flightProvider = flightProvider;
```

Comment:
The `flightProvider` in `MSIDFlightManager` may not be thread safe. Since this is a global instance. While one thread is reading the `flightProvider`, and another SsoExt callback is assigning value here, will cause crash.

Reply:
Thank you Kai, updated.